torch/_inductor/codegen/wrapper.py

```diff
@@ -1124,11 +1136,10 @@ def write_prefix(self):
         )
 
     def define_kernel(
-        self, name: str, kernel: str, metadata: Optional[str] = None, cpp=False
+        self, name: str, kernel: str, metadata: Optional[str] = None, cuda=True
     ):
-        if cpp:
+        if not cuda:
```

Comment:
what's the reason for changing arg from cpp to cuda?

Reply:
The main reason is because `cpp` means two things in inductor's codegen, the cpu backend and the cpp wrapper codegen. Because I added generate_kernel_call for the triton backend into the base class here, WrapperCodeGen, now we need a flag to distinguish the two backends, and `cpp` wouldn't be a good naming here. `cuda` may not be the best naming, but it's slightly better.

Reply:
@desertfire , I'm trying to enable Inductor to support other backends at runtime https://github.com/pytorch/pytorch/pull/100706. The design assumes the backend just needs to customize two meta classes, `BaseScheduling` and `wrapper.WrapperCodeGen`, by overriding some member functions. Regarding the `define_kernel`, I'm kind of confused by the `cuda` parameter because it is the logic of  `CudaWrapperCodeGen` and it has implied `cuda` is `True`. My question is this parameter is must-have.

Reply:
Initially I didn't have `cuda` as a parameter, but the problem appears when we have a mix of devices (see `test_multi_device`). Because we only keep one wrapper codegen instance, i.e. CudaWrapperCodeGen when there is a mix of cuda and cpu tensors, we need this additional parameter to help generate correct kernel definition and kernel call.